csrc/device_lower/pass/index.cpp

```diff
@@ -1742,10 +1743,240 @@ static MmaInputSmemSwizzle getSwizzleMode(TensorView* tv) {
+  for (auto it = alloc_domain.rbegin(); it != alloc_domain.rend(); ++it) {
+    if (!(*it)->front()->as<IterDomain>()->isBroadcast()) {
+      inner = *it;
+      break;
+    }
+  }
+  NVF_ERROR(
+      inner != nullptr,
+      "Matmul with all broadcasting dimension is not supported yet.");
+
+  auto exprs =
+      ValGraphBFS::getExprsBetween(id_graph, loop_domain, alloc_domain);
+  while (!exprs.empty()) {
+    auto [expr, direction] = exprs.back();
+    exprs.pop_back();
+    auto from =
+        (direction == Direction::Backward ? id_graph.inputGroups(expr)
+                                          : id_graph.outputGroups(expr));
+    auto to =
+        (direction == Direction::Backward ? id_graph.outputGroups(expr)
+                                          : id_graph.inputGroups(expr));
+    bool in_from = std::find(from.begin(), from.end(), inner) != from.end();
+    if (!in_from) {
+      continue;
+    }
+    NVF_ERROR(
+        from.back() == inner,
+        "Expecting the innermost of the alloc domain to stay inner");
+    inner = to.back();
+  }
+  IterDomain* inner_id = nullptr;
+  for (auto id : mma->out()->as<TensorView>()->getLoopDomain()) {
+    if (inner->has(id)) {
+      inner_id = id;
+      break;
+    }
+  }
+  NVF_ERROR(
+      inner_id != nullptr,
+      "Could not find innermost ID in the loop domain of mma output");
+  NVF_ERROR(
+      inner_id->getParallelType() == ParallelType::Mma,
+      "Expecting the innermost ID to be parallelized on Mma");
+  return inner;
+}
+
+// Compute the "leading_bytes" in the matrix descriptor of Mma. The leading
+// bytes is the stride of the innermost dimension in the allocation domain of tv
+// considering core matrices. For example, if the tv is [M, K], where K is the
+// inner, then the schedule of the loop domain of the mma output must be
+// something like:
+//      M            K
+//      |            |
+//     ...          ...
+//      |            |
+//     / \.         / \.
+//   ...  m_inst  ...  k_inst
+// where m_inst and k_inst are the instruction tiles of M and K, respectively,
+// that is, the number of items each TensorCore instruction can execute. Both
+// m_inst and k_inst must be parallelized on Mma. The leading_bytes is the
+// stride of the outer (k_inst/swizzle_size) in the allocation domain of tv.
+// That is, if we futher split k_inst as:
+//      M            K
+//      |            |
+//     ...          ...
+//      |            |
+//     / \.         / \.
+//   ...  m_inst  ...  k_inst
+//                     /    \.
+//               linear      swizzle_size
+// Then we would need to prove that `linear` is linear in the allocation domain
+// of tv, and the stride of `linear` is the leading_bytes. This function does
+// the following things:
+// 1. Find k_inst.
+// 2. Split k_inst as above.
+// 3. Prove that `linear` is linear in the allocation domain of tv, and get the
+//    stride of `linear`.
+Val* getInnerStride(TensorView* tv, const MmaOp* mma) {
```

Comment:
Should we name this `getInnerStrideBytes`?

Reply:
Good suggestion!

Reply:
Renamed